AI_ML/CLASSIFICATION/SUPPORT_VECTOR_MACHINE/SUPPORT_VECTOR_MACHINE.py

```diff
@@ -25,40 +28,19 @@ def SUPPORT_VECTOR_MACHINE(
         The predictions for the input data.
     """
 
-    if len(dc_inputs) != 2:
-        raise ValueError(
-            "SUPPORT_VECTOR_MACHINE requires both training data and input data"
-        )
-
-    training_data = dc_inputs[0]
-    input_data = dc_inputs[1]
-
-    if training_data.type != "dataframe" and training_data.type != "matrix":
-        raise ValueError(
-            f"unsupported DataContainer type passed to SUPPORT_VECTOR_MACHINE node for train: {training_data.type}"
-        )
-
-    if input_data.type != "dataframe" and input_data.type != "matrix":
-        raise ValueError(
-            f"unsupported DataContainer type passed to SUPPORT_VECTOR_MACHINE node for input: {input_data.type}"
-        )
-
-    target: str = params["target"]
-    kernel: str = params.get("kernel", "linear")
-
     le = preprocessing.LabelEncoder()
 
-    if training_data.type == "dataframe":
-        df = cast(pd.DataFrame, training_data.m)
+    if isinstance(training_data, DataFrame):
+        df = training_data.m
         if not target:
             target = str(df.columns[-1])
 
         col = df[target]
-        train = cast(pd.DataFrame, df.drop(target, axis=1)).to_numpy()
+        train = df.drop(target, axis=1).to_numpy()
     # Other case is matrix
     else:
-        # assume the last column is the label
-        data = cast(np.ndarray, training_data.m)
+        # assume the last column is the labelxw
```

Comment:
How about supplying `training_labels` in the input as a `DataFrame | Matrix` such that the distinction is clear? Otherwise we have to make assumptions and this is prone to errors and misunderstanding

Reply:
In that case, would it be better to separate the training data into training_label and training_features (extracted directly inside SVM node) and extract training label using `EXTRACT_COLUMNS` node?

Reply:
Great idea. This would be modular, flexible and reusable. It would also make this `SVM` node simpler.